zilliqa/src/precompiles/scilla.rs

```diff
@@ -622,6 +626,7 @@ fn scilla_call_precompile<I: ScillaInspector>(
     };
     trace!(?result, "scilla_call complete");
     if !&result.success {
+        evmctx.db = state;
```

Comment:
This allows failed scilla transitions to modify the state - do we want to allow this?